deepwell/src/services/filter/service.rs

```diff
@@ -91,6 +99,32 @@ impl FilterService {
             ..Default::default()
         };
 
+        // Handle case-sensitivity logic
+        if let ProvidedValue::Set(case_sensitive) = case_sensitive {
+            match regex {
+                // If the regex is being changed, add case-insensitivity flag if case-insensitive.
+                ProvidedValue::Set(ref mut regex) if !case_sensitive => {
+                    regex.insert_str(0, "(?i)")
```

Comment:
I'm under the impression that if someone adds `(?i)` to the start of a case-sensitive regex, they intend to make it case-insensitive. For this reason, it isn't removed if the case-sensitive option is toggled. It will be interpreted by the database as case-insensitive and would be indistinguishable from regularly toggling the case sensitivity.

Reply:
Okay, now that I'm at my desktop and thinking this through, this make sense. For ease of readability, consider a `Provided::Value::Set(_) => (),` branch underneath with a comment for this case (then the `_ => ()` can be removed too).

Reply:
Sounds good to me